xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -7324,7 +7324,11 @@ public void resetArchive(XWikiContext context) throws XWikiException
     {
         boolean hasVersioning = context.getWiki().hasVersioning(context);
         if (hasVersioning) {
-            getVersioningStore(context).resetRCSArchive(this, true, context);
+            // Pre-load the document's archive to prevent hibernate states inconsistencies
+            // when calling resetRCSArchive.
+            this.loadArchive(context);
```

Comment:
As said in a previous comment resetRCSArchive is already taking care of loading the archive when it's not here already as far as I can see in the code. If there is a bug in XWikiHibernateVersioningStore#getXWikiDocumentArchive it seems to me it should be fixed there instead of trying a random thing without any idea of why it fixes it.

Reply:
@manuelleduc do you plan to take it back to fix this?